3DAmsterdam/Assets/Amsterdam3D/Scripts/Interface/Snapshot.cs

```diff
@@ -15,24 +20,241 @@ public class Snapshot : MonoBehaviour
+                snapshotCamera.Render();          
+                
+                // Resets canvas
+                responsiveCanvas.worldCamera = null;
+                responsiveCanvas.renderMode = RenderMode.ScreenSpaceOverlay;
+
+                screenShot.ReadPixels(new Rect(0, 0, width, height), 0, 0);
+                screenShot.Apply();
+
+                // Resets variables
+                snapshotCamera.targetTexture = null;
+                RenderTexture.active = null;
+                snapshotMainMenu.isOn = true;
+
+
+                // If no filetype is given, make it a png
+                if (fileType == "")
+                {
+                    fileType = ".png";
+                }
+
+                // Default filename
+                if (fileName == "")
+                {
+                    fileName = "Screenshot_" + DateTime.Now.ToString("yyyy-MM-dd_HH-mm-ss") + "." + fileType;
+                }
+                else
+                {
+                    fileName = fileName + "." + fileType;
+                }
+
+                byte[] bytes = null;
+                if (fileType == "png")
+                {
+                    bytes = screenShot.EncodeToPNG();
+                }
+                else if (fileType == "jpg")
+                {
+                    bytes = screenShot.EncodeToJPG();
+                }
+                else if (fileType == "raw")
+                {
+                    bytes = screenShot.GetRawTextureData();
+                }
+
+                DownloadFile(bytes, bytes.Length, fileName);
+
+                // Exits out of loop
+                StopCoroutine(screenshotCoroutine);
+                gameObject.SetActive(false);
+            }
+        }
+
+    }
+
+
+#if UNITY_EDITOR
+
+    private IEnumerator ScreenshottingEditor()
+    {
+        while(true)
+        {
+            // Helps with making sure the camera is ready to render
+            yield return new WaitForEndOfFrame();
+            // A trick to make sure the UI gets rendered to the screenshot
+            yield return new WaitForEndOfFrame();
+            if (takeScreenshotOnNextFrame)
+            {
+                takeScreenshotOnNextFrame = false;
+
+                // Creates off-screen render texture that can be rendered into
+                screenshotRenderTexture = new RenderTexture(width, height, 24);
+                screenShot = new Texture2D(width, height , TextureFormat.RGB24, false);
+                    
+                snapshotCamera.targetTexture = screenshotRenderTexture;
+                RenderTexture.active = screenshotRenderTexture;
+
+                // Calls events on the camera related to rendering
+                snapshotCamera.Render();
+
+                // Resets canvas
+                responsiveCanvas.worldCamera = null;
+                responsiveCanvas.renderMode = RenderMode.ScreenSpaceOverlay;
+
+                screenShot.ReadPixels(new Rect(0, 0, width, height), 0, 0);
```

Comment:
Ahh I think I see why the artifacts happen. We never disable the CameraModeChanger.Instance.ActiveCamera, so it is also rendering to the rendertexture (because it is set via RenderTexture.active)
You could eather disable, and later enable the CameraModeChanger.Instance.ActiveCamera after we finished our screenshot, or by making sure the Screenshot camera has a higher priority (and is always rendered on top of the other camera's). To do that you can change the 'priority' field of the Screenshot camera to a value that is higher than the normal camera's.

Reply:
If changing priority fixes the issue then I plan on doing that.